### PR TITLE
Update to poetry@1.1

### LINKED ISSUE #1642: Update to poetry@1.1
We should change:

1. Declaration in `pyproject.toml`

```yaml
[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```
2. How we handle install in https://github.com/wemake-services/wemake-python-styleguide/blob/master/.github/workflows/test.yml Because currently we use outdated setup. See this commit https://github.com/dry-python/returns/commit/0f75fe1d33743ab3b267b7113709c5d9efe45301

### PR DESCRIPTION
**Pull request by serkanozer**
## Checklist

- [X] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [X] I have added my changes to the `CHANGELOG.md`

## Related issues
Closes #1642 
Wasn't able to check locally anything broken on workflow yet, but I just followed
https://python-poetry.org/docs/#installation
https://python-poetry.org/blog/announcing-poetry-1-1-0.html

🙏 Please, if you or your company is finding wemake-python-styleguide valuable, help us sustain the project by sponsoring it transparently on https://opencollective.com/wemake-python-styleguide. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
         pip install -U pip
 
         curl -sSL \
-          "https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py" | python
+          "https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py" | python -
```
Comment (skarzi) [inline .github/workflows/test.yml:24]: Let's add also `poetry` to `${PATH}`:

```bash
# add `poetry` to `${PATH}`
echo "${HOME}/.poetry/bin" >> ${GITHUB_PATH}
```

References:

+ https://python-poetry.org/docs/#osx-linux-bashonwindows-install-instructions
+ https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#adding-a-system-path

Comment (serkanozer) [reply .github/workflows/test.yml:24]: Thanks, it should be fine now I think

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -21,7 +21,9 @@ jobs:
...
+
+        echo "${HOME}/.poetry/bin" >> $GITHUB_PATH
```
Comment (sobolevn) [inline .github/workflows/test.yml:26]: Awesome! Let's also detele all `source "$HOME/.poetry/env"`
They are not needed after this change.

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot! 👍

**Comment by sobolevn** [comment]
@skarzi thanks a lot for the review 🙂

File: .github/workflows/test.yml (hunk 2 above)
Comment (serkanozer) [reply .github/workflows/test.yml:26]: Its there now thanks!

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks a lot! 👍

**Pull request merged**